driver-mapping/src/main/java/com/datastax/driver/mapping/ReflectionUtils.java

```diff
@@ -63,13 +58,15 @@
     // for each key representing a property name,
     // value[0] contains a Field object, value[1] contains a PropertyDescriptor object;
     // they cannot be both null at the same time
-    static <T> Map<String, Object[]> scanFieldsAndProperties(Class<T> baseClass) {
+    static <T> Map<String, Object[]> scanFieldsAndProperties(Class<T> baseClass, HierarchyScanStrategy hierarchyScanStrategy) {
         Map<String, Object[]> fieldsAndProperties = new HashMap<String, Object[]>();
-        Map<String, Field> fields = scanFields(baseClass);
+        List<Class<?>> classesToScan = Lists.<Class<?>>newArrayList(baseClass);
+        classesToScan.addAll(hierarchyScanStrategy.filterClassHierarchy(baseClass));
```

Comment:
Why not require that the strategy include the base class in the list? Then we could avoid that copy.